packages/opentelemetry-exporter-collector/src/transform.ts

```diff
@@ -256,6 +328,209 @@ function toCollectorInstrumentationLibrarySpans(
   };
 }
 
+/**
+ * Given a MetricDescriptor, return its type in a compatible format with the collector
+ * @param descriptor
+ */
+export function toCollectorType(
+  metric: MetricRecord
+): opentelemetryProto.metrics.v1.MetricDescriptorType {
+  const metricKind = metric.descriptor.metricKind;
```

Comment:
```typescript
const { metricKind, valueType } = metric.descriptor;
```

Reply:
This is likely a hot path, so I would say this allocation is actually probably not needed. I would just change all uses of `metricKind` to `metric.descriptor.metricKind` and so on.